navapbc/wic-mt-demo-project-eligibility-screener
# Refactor (4/8): Simple component changes

## Ticket

- https://wicmtdp.atlassian.net/browse/WMDP-253
- https://wicmtdp.atlassian.net/browse/WMDP-254
- https://wicmtdp.atlassian.net/browse/WMDP-257

Came out of refactor work tied to:
- https://wicmtdp.atlassian.net/browse/WMDP-236

## Changes
> What was added, updated, or removed in this PR.

- Update the Accordion, Alert, BackLink, ClinicInfo, List, ReviewElement, and StyledLink components to:
  - Make types consistent
  - Update stories
  - Add tests with snapshots

## Context for reviewers
> Testing instructions, background context, more in-depth details of the implementation, and anything else you'd like to call out or ask reviewers. Explain how the changes were verified.

### Stacked PR Overview

This is part of a series of stacked PRs that break down an overly-large chunk of work that originated in refactoring work. The original branch and draft PR is https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/pull/136  

These stacked PRs are not semantic PRs (i.e. they do not contain work organized by feature) as that would have been very difficult to recreate out of the overly-large chunk of work. Instead, these PRs are broken down to accomplish the following goals:

1. Make the PRs small enough to be reviewable
2. Group similar work together to make it easy to review
3. Commit dependencies first (i.e. low number PRs have dependencies and packages, high number PRs have complex components and pages)
4. Make each PR pass CI automated testing
5. Make each file that was changed only included in a single PR to reduce possibility for merge/rebase conflicts

### Recommended PR Review Strategy

The commits in each PR have been structured to group similar work together, so I recommend using Github's PR "commits" tab to review each PR commit-by-commit. 

### Stacked PR 4/8 

This PR (4 of 8) handles "simple" component changes. Each component in this PR was updated using the following pattern:

- Name and export the component props (e.g. `AccordionProps` for the `Accordion` component)
- Use the new shared types declared in `/src/types.ts` (added in https://github.com/navapbc/wic-mt-demo-project-eligibility-screener/pull/141)
- Use `type` instead of `interface` unless explicitly intending to `extend` the interface
- Use a consistent way of creating storybook stories based on [their typescript examples](https://storybook.js.org/docs/react/get-started/whats-a-story)
- Add unit tests for each component that includes a default snapshot test and exercises all prop options

## Testing
> Screenshots, GIF demos, code examples or output to help show the changes working as expected. ProTip: you can drag and drop or paste images into this textbox.

These changes can be tested using...

1. Install the packages with `yarn install`
5. Run the automated tests with `yarn test`

... and also by running storybook:

1. Build the docker image and start the container: `docker-compose up -d --build`
2. Run storybook in the docker container: `docker-compose exec nextjs yarn storybook`
3. Navigate to `localhost:6006`
4. Play around with the components 
5. Shut down the container with `docker-compose down`
